retry/get_body_go17.go

```diff
@@ -11,6 +11,13 @@ import (
 
 func getBody(r *http.Request) func() (io.ReadCloser, error) {
 	if r.Body != nil {
+		// Optimise for io.ReadSeeker (e.g file readers) for uploading large files.
```

Comment:
While there, perhaps we should also handle the case when `r.Body` is `io.ReadCloser`?

Reply:
Discussed offline, obviously `Body` is always `io.ReadCloser`. The purpose of this function is to make it possible to read the `Body` multiple times, similar to what https://golang.org/src/net/http/request.go?s=33228:33279#L191 is for.  @ht290 I think it's worth documenting it on this function.

Reply:
Added documentation :)